clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/TriggerAdapter.kt

```diff
@@ -0,0 +1,220 @@
+
+        val operator = property.optTriggerOperator(Constants.INAPP_OPERATOR)
+
+        return TriggerCondition(
+            property.optString(Constants.INAPP_PROPERTYNAME, ""),
+            operator,
+            value
+        )
+    }
+
+    /**
+     * Retrieve a TriggerCondition at the specified index from event properties.
+     *
+     * @param index The index of the TriggerCondition to retrieve.
+     * @return The TriggerCondition at the specified index or null if not found or invalid index.
+     */
+    fun propertyAtIndex(index: Int): TriggerCondition? {
+        if (properties.isInvalidIndex(index)) {
+            return null
+        }
+
+        val propertyJSONObject = properties?.optJSONObject(index) ?: return null
+
+        return triggerConditionFromJSON(propertyJSONObject)
+    }
+
+
+    /**
+     * Retrieve a TriggerCondition at the specified index from item properties.
+     *
+     * @param index The index of the TriggerCondition to retrieve.
+     * @return The TriggerCondition at the specified index or null if not found or invalid index.
+     */
+    fun itemAtIndex(index: Int): TriggerCondition? {
+        if (items.isInvalidIndex(index)) {
+            return null
+        }
+
+        val itemJSONObject = items?.optJSONObject(index) ?: return null
+
+        return triggerConditionFromJSON(itemJSONObject)
+    }
+
+    /**
+     * Retrieves a [TriggerGeoRadius] instance at the specified index from the geoRadiusArray.
+     *
+     * The function checks if the provided index is valid and, if so, extracts the latitude, longitude, and radius
+     * information from the corresponding JSON object within the geoRadiusArray. It then creates and returns a
+     * [TriggerGeoRadius] instance based on the extracted data.
+     *
+     * @param index The index indicating the position of the desired [TriggerGeoRadius] in the geoRadiusArray.
+     * @return A [TriggerGeoRadius] instance if the index is valid and data is available; otherwise, `null`.
+     */
+    fun geoRadiusAtIndex(index: Int): TriggerGeoRadius? {
+        if (geoRadiusArray.isInvalidIndex(index)) {
+            return null
+        }
+
+        val geoRadiusItem = geoRadiusArray?.optJSONObject(index) ?: return null
+
+        val latitude = geoRadiusItem.optDouble("lat")
+        val longitude = geoRadiusItem.optDouble("lng")
+        val radius = geoRadiusItem.optDouble("rad")
+        return TriggerGeoRadius(latitude, longitude, radius)
+    }
+
+    /**
+     * Converts the TriggerAdapter instance into a JSON object.
+     *
+     * The function creates a JSON object representing the TriggerAdapter instance, including the event name, event properties,
+     * item properties, and geographical radius properties (if available). The resulting JSON object is suitable for further
+     * serialization or transmission.
+     *
+     * @return A [JSONObject] representing the TriggerAdapter instance.
+     */
+    fun toJsonObject(): JSONObject {
```

Comment:
rather than recreating this object we should store the val we passed in constructor as is and return it.

Reply:
@shivamsharma2710

Reply:
This is done. Please close the conversation.